operator/pkg/ingress/ingress.go

```diff
@@ -38,6 +38,7 @@ type ingressReconciler struct {
 	defaultLoadbalancerMode string
 	defaultSecretNamespace  string
 	defaultSecretName       string
+	enforcedHTTPS           bool
```

Comment:
@youngnick looks `enforcedHTTPS` isn't handled in `newIngressReconciler` - hence it's always `false`. We should pass the corresponding config value from the cell.

I opened https://github.com/cilium/cilium/pull/30804